### DISCUSSION TITLE
persist_mode not working

### DISCUSSION THREAD

**Discussion by marco-calautti** [rustdesk/rustdesk#9441] [General]
### Bug Description

Everytime I try to connect to my KDE-Wayland machine, it asks for permission to share the screen. Then, it does not ask again until I reboot the Wayland machine.

### How to Reproduce

Simply connect to a Linux machine running Wayland

### Expected Behavior

The permission should be asked only once, and persist_mode should be used to remember the choice.

### Operating system(s) on local side and remote side

Arch with Plasma 6 + Wayland

### RustDesk Version(s) on local side and remote side

1.3.0

### Screenshots

![Schermata_20240923_133819](https://github.com/user-attachments/assets/96a2a855-f8a1-4a83-8e19-bc5efdadf065)

**Comment by fufesou** [comment]
@marco-calautti Hi, thanks for your feedback.

I've tried KDE Plasma 6, "persist mode" works fine.

Can you please run the following command on the controlled side?

1. `pacman -Qi xdg-desktop-portal`. Mine shows Version `1.18.4-1`
2. `grep 'wayland-restore-token' ~/.config/rustdesk/RustDesk_local.toml`
3. Check if there're some errors after selecting the monitor, log file `~/.local/share/logs/RustDesk/server/Rustdesk_rCURRENT.log`

**Comment by xlionjuan** [comment]
You should know it is the situation of the Wayland...

**Reply by andrejpan** [reply]
I am noticing a similar problem. What is the meaning of "it is the situation"?

Also, could I somehow select the screen via SSH access?

**Reply by xlionjuan** [reply]
Unattended Access on Wayland still ***NOT*** mature, please use X11 if you want unattended access 


No, please use Zerotier/Tailscale or Cloudflare ZeroTrust+Tunnel, it has web SSH.

**Reply by andrejpan** [reply]
What is the meaning of "not mature"? Could I disable it already in some config file?

Although, I set it to log in with X11, the newest Ubuntu, I sometimes see that it boots to Wayland.

In what sense are Zerotier/Tailscale solving this situation if I would like to have GUI access to a remote machine? Waland would still block them, right?

**Reply by xlionjuan** [reply]
> What is the meaning of "not mature"? Could I disable it already in some config file?

Just ensure you enter X11 environments, for Ubuntu, choose "Ubuntu with x.org" or similar meaning entity 

> Although, I set it to log in with X11, the newest Ubuntu, I sometimes see that it boots to Wayland.

It will remember 

> In what sense are Zerotier/Tailscale solving this situation if I would like to have GUI access to a remote machine? Waland would still block them, right?

On Linux, everything, every components are separated, your SSH will works no matter what window system/server you're using 




It seems you're newbie to the Linux, Wiki and YouTube are your good friends.

**Reply by andrejpan** [reply]
> It seems you're newbie to the Linux, Wiki and YouTube are your good friends.

I appreciate your advice, but there's no need to patronize. Everyone is at different stages of learning, and respectful guidance goes a long way in fostering a positive environment. Let's keep the conversation constructive.

Again, I asked what the benefits of Zerotier/Tailscale tool for GUI access compared with Rustdesk? How are they bypassing Wayland, or do they have their own protocol for transferring GUI?

**Reply by xlionjuan** [reply]
Sorry, I mess seeing "screen" in your comment, I will answer again

> Also, could I somehow select the screen via SSH access?

Technically, yes, there is thing called "X11 forwarding", but it is very old technology and should avoid using it due to security reasons.

You could still use Zerotier/Tailscale to link to your machine though SSH(command line only), if you need GUI, and you are not willing to use X11, your only choice is the Remote Desktop that comes with your DE, such as GNOME and KDE.

And unfortunately...again, it is not easy, you'd better use x11 if you need this.

**Comment by marco-calautti** [comment]
To be honest, the main point of a *remote control* application like rustdesk is to not have to confirm access to the screen on the host machine everytime . This is not *not mature*, it is simply unusable.

Wayland support in this state is like not having Wayland support at all.

**Reply by xlionjuan** [reply]
It is Wayland's problems unfortunately.....


The Linux has lot's of fragmentation, but things makes better more quickly in recent years thanks for Microsoft (for doing sxxt) and Valve.
